java/ingest/ingest-batcher-submitter/src/main/java/sleeper/ingest/batcher/submitter/IngestBatcherSubmitter.java

```
@@ -42,20 +44,30 @@ public class IngestBatcherSubmitter {
     private final Configuration conf;
     private final TableIndex tableIndex;
     private final IngestBatcherStore store;
+    private final IngestBatcherSubmitDeadLetterQueue dlQueue;
 
-    public IngestBatcherSubmitter(InstanceProperties properties, Configuration conf, TableIndex tableIndex, IngestBatcherStore store) {
+    public IngestBatcherSubmitter(InstanceProperties properties, Configuration conf, TableIndex tableIndex, IngestBatcherStore store,
+            IngestBatcherSubmitDeadLetterQueue dlQueue) {
         this.properties = properties;
         this.conf = conf;
         this.tableIndex = tableIndex;
         this.store = store;
+        this.dlQueue = dlQueue;
     }
 
     public void submit(IngestBatcherSubmitRequest request, Instant receivedTime) {
         List<IngestBatcherTrackedFile> files;
         try {
             files = toTrackedFiles(request, receivedTime);
-        } catch (RuntimeException e) {
-            LOGGER.warn("Error when processing request: {}", request, e);
+        } catch (UncheckedIOException uioe) {
+            if (uioe.getCause() instanceof FileNotFoundException) {
+                LOGGER.info("File not found, sending request: {} to dead letter queue", request, uioe);
+                dlQueue.submit(request);
+            }
+            return;
```

This return statement will discard and ignore any UncheckedIOException that was not caused by a FileNotFoundException. This means if there's a network failure finding the files to add to the store, the exception will be ignored.

The exception needs to be rethrown here. We could really do with testing this case specifically. I'm not sure how we could provide test coverage of this behaviour other than to use WireMock.

---

Re-thrown UncheckedIOException

---

Raised issue #4806 to develop a test to capture the network failure scenario.